src/mol-plugin-state/manager/focus-camera/orient-axes.ts

```diff
@@ -0,0 +1,247 @@
+
+    coords = selectHeavyCoords(structs);
+    if (coords.length >= 3 * minAtoms) return coords;
+
+    coords = selectAllCoords(structs);
+    return coords;
+}
+
+/** Select coordinates of C-alpha and O3' atoms */
+function selectCACoords(structs: Structure[]): number[] {
+    const coords: number[] = [];
+    for (const struct of structs) {
+        for (const unit of struct.units) {
+            const { x, y, z } = unit.conformation;
+            for (let i = 0; i < unit.polymerElements.length; i++) {
+                const index = unit.polymerElements[i];
+                coords.push(x(index), y(index), z(index));
+            }
+        }
+    }
+    return coords;
+}
+
+/** Select coordinates of non-hydrogen atoms, excluding water */
+function selectHeavyCoords(structs: Structure[]): number[] {
+    const coords: number[] = [];
+    for (const struct of structs) {
+        for (const unit of struct.units) {
+            const { x, y, z } = unit.model.atomicConformation;
+            for (let i = 0; i < unit.elements.length; i++) {
+                const index = unit.elements[i];
+                const compound = unit.model.atomicHierarchy.atoms.label_comp_id.value(index);
+                const element = unit.model.atomicHierarchy.atoms.type_symbol.value(index);
+                if (element !== 'H' && compound !== 'HOH') {
+                    coords.push(x[index], y[index], z[index]);
+                }
+            }
+        }
+    }
+    return coords;
+}
+
+/** Select coordinates of all atoms */
+function selectAllCoords(structs: Structure[]): number[] {
+    const coords: number[] = [];
+    for (const struct of structs) {
+        for (const unit of struct.units) {
+            const { x, y, z } = unit.model.atomicConformation;
+            for (let i = 0; i < unit.elements.length; i++) {
+                const index = unit.elements[i];
+                coords.push(x[index], y[index], z[index]);
+            }
+        }
+    }
+    return coords;
+}
+
+/** Return a flip around XYZ axes which minimizes the difference between flip*rotation and referenceRotation. */
+function minimalFlip(rotation: Mat3, referenceRotation: Mat3): Mat3 {
+    let bestFlip = ROTATION_MATRICES.eye;
+    let bestScore = 0; // there will always be at least one positive score
+    const aux = Mat3();
+    for (const flip of [ROTATION_MATRICES.eye, ROTATION_MATRICES.rotX180, ROTATION_MATRICES.rotY180, ROTATION_MATRICES.rotZ180]) {
+        const score = mat3Dot(Mat3.mul(aux, flip, rotation), referenceRotation);
+        if (score > bestScore) {
+            bestFlip = flip;
+            bestScore = score;
+        }
+    }
+    return bestFlip;
+}
+
+/** Measure of how similar two rotation matrices are (dot product of flattened matrices) */
+function mat3Dot(a: Mat3, b: Mat3) {
+    return a[0] * b[0] + a[1] * b[1] + a[2] * b[2] + a[3] * b[3] + a[4] * b[4] + a[5] * b[5] + a[6] * b[6] + a[7] * b[7] + a[8] * b[8];
+}
```

Comment:
Add this to the `Mat3` namespace instead please.

Reply:
Done, and I found out there is an actual name for this operator :D